src/Core/DocumentEvents.cs

```diff
@@ -22,40 +22,39 @@
 
 namespace SonarLint.VisualStudio.Core;
 
-public abstract class DocumentEventArgs(string fullPath,  IEnumerable<AnalysisLanguage> detectedLanguages) : EventArgs
+public class DocumentEventArgs(Document document) : EventArgs
 {
-    /// <summary>
-    /// Full path of the document for which the event was raised
-    /// </summary>
-    public string FullPath { get; } = fullPath;
-    public IEnumerable<AnalysisLanguage> DetectedLanguages { get; } = detectedLanguages;
+    public Document Document { get; } = document;
 }
 
-public class DocumentClosedEventArgs(string fullPath, IEnumerable<AnalysisLanguage> detectedLanguages) : DocumentEventArgs(fullPath, detectedLanguages);
-
-public class DocumentOpenedEventArgs(string fullPath, IEnumerable<AnalysisLanguage> detectedLanguages) : DocumentEventArgs(fullPath, detectedLanguages);
-
-public class DocumentSavedEventArgs(string fullPath, string newContent, IEnumerable<AnalysisLanguage> detectedLanguages) : DocumentEventArgs(fullPath, detectedLanguages)
+public class DocumentSavedEventArgs(Document document, string newContent) : DocumentEventArgs(document)
 {
     public string NewContent { get; } = newContent;
 }
 
-public class DocumentRenamedEventArgs(string fullPath, string oldFilePath, IEnumerable<AnalysisLanguage> detectedLanguages) : DocumentEventArgs(fullPath, detectedLanguages)
+public class DocumentRenamedEventArgs(Document document, string oldFilePath) : DocumentEventArgs(document)
 {
     public string OldFilePath { get; } = oldFilePath;
 }
 
+public class Document(string fullPath, IEnumerable<AnalysisLanguage> detectedLanguages)
+{
+    public string FullPath { get; } = fullPath;
+    public IEnumerable<AnalysisLanguage> DetectedLanguages { get; } = detectedLanguages;
+}
+
 /// <summary>
 /// Raises notifications about document events
 /// </summary>
 public interface IDocumentEvents
```

Comment:
Suggestion: rename to DocumentTracker, since it also returns the list of currently open documents